process/transaction/shardProcess.go

```diff
@@ -195,6 +195,8 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction) (vmco
 		txProc.pubkeyConv,
 	)
 
+	defer txProc.accounts.SetTxHashForLatestStateChanges(txHash)
```

Comment:
The idea was to link all the changes triggered by a tx to the txHash after the tx was executed. It can be refactored as you say, but I would do that in the next PR to avoid generating more conflicts with @ssd04's PR

Reply:
Added TODO